uui-components/src/table/DataTable.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import {
+    IEditable, DataTableState, DataSourceListProps, DataTableColumnsConfigOptions, DataRowProps, DataColumnProps,
+    uuiMarkers, useTableShadows, cx, useVirtualList, IHasCX,
+} from '@epam/uui';
+import { PositionValues, ScrollBars } from '@epam/uui-components';
+import * as css from './DataTable.scss';
+
+export interface DataTableProps<TItem, TId> extends IHasCX, IEditable<DataTableState>, DataSourceListProps, DataTableColumnsConfigOptions {
+    getRows(): DataRowProps<TItem, TId>[];
+    columns: DataColumnProps<TItem, TId>[];
+    renderRow?(props: DataRowProps<TItem, TId>): React.ReactNode;
+    renderNoResultsBlock?(): React.ReactNode;
+    renderTopShadow?: boolean;
+    renderBottomShadow?: boolean;
+    onScroll?(value: PositionValues): void;
+    showColumnsConfig?: boolean;
+    onConfigurationButtonClick?: () => void;
+    renderHeader?: () => React.ReactNode;
+};
+
+enum scrollShadowsCx {
+    top = 'uui-scroll-shadow-top',
+    topVisible = 'uui-scroll-shadow-top-visible',
+    bottom = 'uui-scroll-shadow-bottom',
+    bottomVisible = 'uui-scroll-shadow-bottom-visible'
+};
+
+export function DataTable<TItem, TId>({
+    rowsCount,
+    value,
+    onValueChange,
+    onScroll,
+    columns,
+    ...props
+}: DataTableProps<TItem, TId>) {
+    const { listRef, scrollbarsRef, offsetY, handleScroll, estimatedHeight } = useVirtualList<HTMLDivElement>({
```

Comment:
move useVirtualLust hook usage and shadows logic to the VirtualList component. Remove DataTable from uui-components, just leave in skins